sources/Interop/Windows/winrt/wrl/client/ComPtr`1.Manual.cs

```diff
@@ -4,150 +4,401 @@
         {
-            fixed (T** ptr = &ptr_)
+            fixed (T** p = &ptr_)
             {
-                return ref *ptr;
+                return ref *p;
             }
         }
 
+        /// <summary>
+        /// Releases the current COM object in use and gets the address of the <see cref="ComPtr{T}"/> instance as a raw <typeparamref name="T"/> double pointer.
+        /// This method is only valid when the current <see cref="ComPtr{T}"/> instance is on the stack or pinned.
+        /// </summary>
+        /// <returns>The raw pointer to the current <see cref="ComPtr{T}"/> instance.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public T** ReleaseAndGetAddressOf()
         {
             InternalRelease();
+
             return GetAddressOf();
         }
 
+        /// <summary>
+        /// Resets the current instance by decrementing the reference count for the target COM object and setting the internal raw pointer to <see langword="null"/>.
+        /// </summary>
+        /// <returns>The updated reference count for the COM object that was in use, if any.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public uint Reset()
         {
             return InternalRelease();
         }
 
-        public void Swap(ref ComPtr<T> r)
+        /// <summary>
+        /// Swaps the current COM object reference with that of a given <see cref="ComPtr{T}"/> instance.
+        /// </summary>
+        /// <param name="r">The target <see cref="ComPtr{T}"/> instance to swap with the current one.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void Swap(ComPtr<T>* r)
         {
-            T* tmp = ptr_;
-            ptr_ = r.ptr_;
-            r.ptr_ = tmp;
+            T* temp = ptr_;
+
+            ptr_ = r->ptr_;
+
+            r->ptr_ = temp;
         }
 
+        /// <summary>
+        /// Swaps the current COM object reference with that of a given <see cref="ComPtr{T}"/> instance.
+        /// </summary>
+        /// <param name="other">The target <see cref="ComPtr{T}"/> instance to swap with the current one.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void Swap(ref ComPtr<T> other)
+        {
+            T* temp = ptr_;
+
+            ptr_ = other.ptr_;
+
+            other.ptr_ = temp;
+        }
+
+        // Increments the reference count for the current COM object, if any
         private readonly void InternalAddRef()
         {
-            if (ptr_ != null)
+            T* pointer = ptr_;
+
+            if (pointer != null)
             {
-                ((IUnknown*)(ptr_))->AddRef();
+                ((IUnknown*)pointer)->AddRef();
             }
         }
 
+        // Decrements the reference count for the current COM object, if any
         private uint InternalRelease()
         {
-            uint @ref = 0;
+            uint referenceCount = 0;
```

Comment:
The names here were matching the ones used in native code. It would be good to keep them where possible. Same for other places as well where changes have been made to the code.

Reply:
I see, that makes perfect sense 😄
Reverted those changes in 3f93d6eea94a5528d3931543689c401c03289d07.